apiary.apib

```diff
@@ -1738,6 +1738,699 @@ root will not be listable, as it is virtual. Access known paths directly.
+        v0.4.0-dev
+        ```
+
+## install [GET /update/install{?arg}]
+Install a version of IPFS.
+
++ Parameters
+    + arg (string, required) - The IPFS version to install.
+
++ Request Without Arguments
+
+    The response is the same if the argument is empty. For example:
+
+        curl -i "http://localhost:5001/api/v0/update/install?arg="
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/update/install"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/update/install"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: text/plain
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Stream-Output: 1
+        Date: Thu, 11 Feb 2016 22:23:28 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        ERROR: Please specify a version to install
+        ```
+
++ Request With Invalid Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/update/install?arg=kitten"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/update/install?arg=kitten"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: text/plain
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Stream-Output: 1
+        Date: Thu, 11 Feb 2016 22:26:07 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        fetching ipfs version vkitten
+        install failed, reverting changes...
+        ERROR: no link named "vkitten" under QmdvSvgDA2C64eGcRKnWiBwZCdM6WCKSMa38d1hsvMkcRZ 
+        ```
+
++ Request With Valid Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/update/install?arg=v0.3.5"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/update/install?arg=v0.3.5"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: text/plain
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Stream-Output: 1
+        Date: Thu, 11 Feb 2016 22:26:07 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        fetching ipfs version v0.3.5
+        binary downloaded, verifying...
+        install failed, reverting changes...
+        ERROR: version didnt match 
```

Comment:
This looks like it failed, would be good to have an example of an actually sucessfull update